packages/trader/src/Modules/Trading/Components/Elements/mobile-widget.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { Money } from '@deriv/components';
+import { localize, Localize } from '@deriv/translations';
+import { getExpiryType, getDurationMinMaxValues, getLocalizedBasis } from '@deriv/shared';
+import { MultiplierAmountWidget } from 'Modules/Trading/Components/Form/TradeParams/Multiplier/widgets.jsx';
+import TradeParamsModal from '../../Containers/trade-params-mobile.jsx';
+import { observer, useStore } from '@deriv/stores';
+import { useTraderStore } from 'Stores/useTraderStores';
+
+const MobileWidget = observer(
+    ({ toggleDigitsWidget, is_collapsed }: { toggleDigitsWidget: () => void; is_collapsed: boolean }) => {
+        const { ui } = useStore();
+        const { onChangeUiStore } = ui;
+        const trade_store = useTraderStore();
+        const {
+            amount,
+            basis,
+            currency,
+            duration,
+            duration_min_max,
+            duration_unit,
+            form_components,
+            is_multiplier,
+            last_digit,
+            onChange,
+        } = trade_store;
+
+        const [is_open, setIsOpen] = React.useState(false);
+
+        React.useEffect(() => {
+            assertDurationIsWithinBoundary();
+        });
+
+        const assertDurationIsWithinBoundary = () => {
+            const contract_expiry_type = getExpiryType(trade_store);
+            const [min_value, max_value] = getDurationMinMaxValues(
+                duration_min_max,
+                contract_expiry_type,
+                duration_unit
+            );
+
+            if (contract_expiry_type === 'tick' && min_value !== null && duration < min_value) {
+                onChangeUiStore({ name: `duration_${duration_unit}`, value: min_value });
+                onChange({ target: { name: 'duration', value: min_value } });
+            }
+            if (min_value !== null && max_value !== null && !(duration < min_value) && duration > max_value) {
+                onChangeUiStore({ name: `duration_${duration_unit}`, value: max_value });
+                onChange({ target: { name: 'duration', value: max_value } });
+            }
+        };
+
+        const toggleWidget = () => setIsOpen(!is_open);
+
+        const getHumanReadableDuration = () => {
+            if (!duration_unit) return '';
+            const lookup = {
+                t: [localize('tick'), localize('ticks')],
+                s: [localize('second'), localize('seconds')],
+                m: [localize('min'), localize('mins')],
+                h: [localize('hour'), localize('hours')],
+                d: [localize('day'), localize('days')],
+            };
+
+            try {
+                if (!duration_unit) return '';
+                const formatted_duration_unit =
+                    +duration === 1
+                        ? lookup[duration_unit as keyof typeof lookup][0]
```

Comment:
duration unit has `string` type and it's used as a key in lookup (where key should be 't'/'s'/'m'/'h'/'d')